### ISSUE TITLE
Fenix hangs while scrolling in history

### ISSUE THREAD

**Issue by cadeyrn** [mozilla-mobile/fenix#881]
### Steps to reproduce

1. Fenix > Menu > Your Library > History
2. scroll down

### Expected behavior

smooth scrolling, no hangs.

### Actual behavior

After scrolling a bit Fenix hangs for a few seconds and doesn't react to anything. You can try to scroll, but nothing happens. After the hang you can scroll a bit, but then there is the next hang for a few seconds, and so on.

This makes the history view unusable.

### Device information

* Android device: HTC U11 / Android 8.0
* Fenix version: Fenix revision 8b793cd43ea0b816fece4266c2dbc990e24ceae1

**Comment by vesta0** [comment]
@sv-ohorvath please verify that this is still happening.

**Comment by cadeyrn** [comment]
I can no longer reproduce since I cleared all app data a week or two weeks ago. It might depend on the contents of the history. That's only a theory but I guess there were a few very long data URIs of the "About Fenix" page in my history (from before it was implemented as an Android Fragment) because I tested a lot with "About Fenix", so maybe related to #1162 and https://github.com/mozilla-mobile/android-components/issues/2534 - it's about the awesomebar suggestions but I guess the same applies for the history view as well.
